docs/timezone.md

```diff
@@ -32,10 +87,37 @@ However, most other code will assume that custom time zones act like built-in `T
 To interoperate with libraries or other code that you didn't write, then you should implement all the other `Temporal.TimeZone` members as well: `id`, `getOffsetStringFor()`, `getPlainDateTimeFor()`, `getInstantFor()`, `getNextTransition()`, `getPreviousTransition()`, and `toJSON()`.
 Your object must not have a `timeZone` property, so that it can be distinguished in `Temporal.TimeZone.from()` from other Temporal objects that have a time zone.
 
-The identifier of a custom time zone must consist of one or more components separated by slashes (`/`), as described in the [tzdata documentation](https://htmlpreview.github.io/?https://github.com/eggert/tz/blob/master/theory.html#naming).
-Each component must consist of between one and 14 characters.
-Valid characters are ASCII letters, `.`, `-`, and `_`.
-`-` may not appear as the first character of a component, and a component may not be a single dot `.` or two dots `..`.
+### Custom time zone identifiers
+
+Identifiers of custom time zones must follow the rules described in the [tzdata documentation](https://htmlpreview.github.io/?https://github.com/eggert/tz/blob/master/theory.html#naming):
+
+- A valid identifier has one or more components separated by slashes (`/`)
+- Each component must consist of between one and 14 characters.
+- Valid characters are ASCII letters, `.`, `-`, and `_`.
+- `-` may not appear as the first character of a component, and a component may not be a single dot `.` or two dots `..`.
+
+### Canonicalization of custom time zone identifiers
+
+Custom time zones must return their canonical string identifier in the `id` getter, so that other code can compare the `id` to know if two time zones are equivalent.
+
+If a custom time zone is intended to be treated as different from all built-in time zones, then its `id` must not match any built-in time zone's canonical identifier.
+
+However, a custom time zone that is intended to be equivalent to a particular built-in time zone must return the same `id` as the corresponding built-in time zone.
+Because canonical identifiers can vary across time and across ECMAScript implementations, that `id` should be read and cached using `Temporal.TimeZone.from` at runtime instead of being coded as a constant:
+
+```javascript
+// Get and cache the canonical identifier for time zones in India
+const indianTimeZoneId = Temporal.TimeZone.from('Asia/Kolkata').id;
+
+class CustomIndianTimeZone {
+  get id() {
+    return indianTimeZoneId;
+  }
+  // DON'T USE A CONSTANT LIKE THIS:  get id() { return 'Asia/Kolkata'; }
+
+  . . . (more implementation code goes here)
+}
```

Comment:
This part doesn't make sense to me; I'd think that if you are polyfilling a builtin time zone, you would want to use the canonical identifier given by the version of the TZDB that you're polyfilling, not the canonical identifier given by the implementation's version of the TZDB. (I think we could actually just skip this part.)

Reply:
I guess that it comes down to whether the calendar author wants their polyfill to be considered equivalent to the built-in zone or not, for example in `ZDT.p.equals`.  The case I had in mind was a "ZDT polyfiller" function that accepted a ZDT and returned a new ZDT which, if the parameter's `timeZone` was the zone the author wanted to polyfill, replaced its `timeZone` with the author's custom zone.  

If that zone has a different canonical ID in different implementations (regardless of whether the reason is different TZDB versions or an implementation's canonicalization choices like in Chrome) then it's probably best to cache the canonical ID rather than hardcoding it. 

What do you think?

Reply:
Regarding that use case, I'd think that the one thing nobody wants is to have two time zones kicking around with canonically equal IDs but different time zone data. So you'd need to polyfill ZonedDateTime such that any entry point that could store the builtin identifier you're replacing in the internal slot, instead puts your polyfilled TimeZone object in there.